patch-db/src/transaction.rs

```diff
@@ -202,4 +206,20 @@ impl<Parent: DbHandle + Send + Sync> DbHandle for Transaction<Parent> {
         self.updates.append(patch);
         Ok(None)
     }
+
+    async fn lock_all<'a>(
+        &'a mut self,
+        locks: Vec<crate::bulk_locks::LockTargetId>,
+    ) -> Result<crate::bulk_locks::Verifier, Error> {
+        let verifier = Verifier {
+            target_locks: locks.iter().cloned().collect(),
+        };
+        self.locks.push(
+            self.parent
+                .locker()
+                .lock_all(self.id.clone(), locks)
+                .await?,
+        );
+        Ok(verifier)
+    }
```

Comment:
I cannot review this, this will need to be reviewed by @dr-bonez

Reply:
is there a reason this doesn't just take `impl IntoIterator<Item = LockTargetId>`

Reply:
Ok, made changes into another PR https://github.com/Start9Labs/patch-db/pull/33
I would like to know why this is a better solution

Reply:
Merged